pkg/importer/data-processor.go

```diff
@@ -264,16 +264,32 @@ func (dp *DataProcessor) convert(url *url.URL) (ProcessingPhase, error) {
 }
 
 func (dp *DataProcessor) resize() (ProcessingPhase, error) {
-	// Resize only if we have a resize request, and if the image is on a file system pvc.
 	size, _ := getAvailableSpaceBlockFunc(dp.dataFile)
 	klog.V(3).Infof("Available space in dataFile: %d", size)
+	isBlockDev := size >= int64(0)
 	shouldPreallocate := false
-	if dp.requestImageSize != "" && size < int64(0) {
-		var err error
-		klog.V(3).Infoln("Resizing image")
-		shouldPreallocate, err = ResizeImage(dp.dataFile, dp.requestImageSize, dp.availableSpace)
-		if err != nil {
-			return ProcessingPhaseError, errors.Wrap(err, "Resize of image failed")
+	if !isBlockDev {
+		// Unlike block devices, a raw image file on a disk doesn't fill up the
+		// file system space as needed (thus we resize it)
+		// And may be bigger than the available space (sparse).
```

Comment:
doesn't fill up as as needed == sparse?

Reply:
Nope - I mean that QEMU will respect the image file size and not make it bigger as necessary.
[I refactored this bit of code specifically so it requires far fewer comments to explain why things are done](https://github.com/kubevirt/containerized-data-importer/pull/1466/commits/a29e9f9518e83a7ce6bd5920e74e070c713a6667)

Reply:
so now you parse and validate also for dp.requestImageSize != "" ?

Reply:
yes. we can probably just validate it and it will pass after the resize - easier than explaining why it's not strictly necessary to do it any more